core/utils/utils_example_test.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+func ExampleStopChan() {
+	stopCh := make(StopChan)
+
+	work := func(context.Context) {}
+
+	a := func(ctx context.Context, done func()) {
+		defer done()
+		ctx, cancel := stopCh.Ctx(ctx)
+		defer cancel()
+		work(ctx)
+	}
+
+	b := func(ctx context.Context, done func()) {
+		defer done()
+		ctx, cancel := stopCh.CtxCancel(context.WithTimeout(ctx, time.Minute))
+		defer cancel()
+		work(ctx)
+	}
+
+	c := func(ctx context.Context, done func()) {
+		defer done()
+		ctx, cancel := stopCh.CtxCancel(context.WithDeadline(ctx, time.Now().Add(5*time.Minute)))
```

Comment:
Do we really want this to be 5 minutes?  Wouldn't something like 5s be enough, and allow the test to fail faster if for some reason it doesn't work?

Reply:
https://github.com/smartcontractkit/chainlink/pull/9050